docs/en/developer/create-web-component/VUE.md

```diff
@@ -1,137 +1,448 @@
+    }
+
+- }
+```
+
+It should look something like this:
+
+``` gradle
+    apply plugin: 'java'
+    apply plugin: 'maven'
+
+    def jsonFile = file("${projectDir}/package.json")
+    def parsedJson = new groovy.json.JsonSlurper().parseText(jsonFile.text)
+    project.version = parsedJson.version + '-SNAPSHOT'
+
+    jar {
+        baseName "uportal__${project.name}"
+        from '.'
+        exclude 'build'
+        exclude 'node_modules'
+        into "META-INF/resources/webjars/uportal__${project.name}/${project.version}"
+    }
+```
+
+**Note:** to produce the most compact .jar possible to reduce traffic on the
+resource server and speed up browser response, or if you are building on
+Windows, try this [build.gradle for Linux and Mac OS](#buildgradle-for-linux-and-mac-os)
+or this [build.gradle for Windows](#buildgradle-for-windows).
+
+### Add Gradle wrapper to project
+
+Run this:
+
+```
+gradle wrapper --gradle-version=5.1.1
+```
+
+Note: now that the project has a build.gradle file and a Gradle wrapper, an
+IDE like IntelliJ IDEA will recognize it as a Gradle project. You could do
+the rest of the required editing in the IDE.
+
+## 4. Edit the Vue application
+
+### Rename HelloWorld.vue
+
+Rename the generated HelloWorld.vue file, replace **{component-name}** with yours:
+
+```
+FROM:
+src/components/HelloWorld.vue
+
+TO:
+src/components/{component-name}.vue
+```
+
+### Edit App.vue
+
+Rename the imports.generated HelloWorld.vue file, replace **{component-name}** with yours:
+
+``` javascript
+// FROM:
+import HelloWorld from './components/HelloWorld.vue'
+
+// TO:
+import HelloWorld from './components/{component-name}.vue'
 ```
 
-### Edit _package.json_
+### Edit package.json
+
+The `{component-name}` after `--name` _**must**_ have a
+hyphen, for example `--name weather-thingy`. Change the following:
+
+```
+// FROM:
+    "build": "vue-cli-service build",
+
+// TO (be sure to rename {component-name}):
+    "prebuild": "babel node_modules/@vue/web-component-wrapper/dist/vue-wc-wrapper.js -o node_modules/@vue/web-component-wrapper/dist/vue-wc-wrapper.js",
+    "build": "vue-cli-service build --name {component-name} --target wc src/components/{component-name}.vue",
+
+```
 
-Update the `scripts` section to generate a web component automatically.
+Add these top-level declarations:
 
-```diff
-- "build": "vue-cli-service build",
-+ "prebuild": "babel node_modules/@vue/web-component-wrapper/dist/vue-wc-wrapper.js -o node_modules/@vue/web-component-wrapper/dist/vue-wc-wrapper.js",
-+ "build": "vue-cli-service build --name {component-name} --target wc src/components/{component-name}.vue",
+```
+  "main": "dist/{component-name}.js",
+  "source": "src/components/{component-name}.vue",
 ```
 
-> **NOTE:** The `prebuild` script transpiles an internal Vue library for Internet Explorer support.
+For example:
 
-Add top-level `main` and `source` attributes.
+```
+{
+  "name": "weather-thingy",
```

Comment:
```suggestion
  "name": "@uportal/weather-thingy",
```

Reply:
Needs to have `@uportal/` if it's supposed to have an `artifactId` starting with `uportal__`, but probably shouldn't if it isn't.

This is something we've omitted previously (and paid a price troubleshooting), so it's great to mention it in the docs.

Reply:
I'm still new to npm and it's nuances. Would adding the '@uportal' make this is scoped package? Would it be preferable to omit this from the instructions to prevent someone publishing a web-component to npm central under this scope? It might be preferable to prevent the use of this package scope except in the case of vetted and approved web components that the community would like to make available to everyone.

Reply:
> Would adding the `@uportal` make this is scoped package?

Yep, that becomes a scoped package.

> Would it be preferable to omit this from the instructions to prevent someone publishing a web-component to npm central under this scope?

Currently publish access to the `@uportal` scope is limited to uPortal committers, https://github.com/Jasig/uPortal/blob/master/docs/COMMITTERS.md
That should limit the risk of accidentally publishing a package.

> It might be preferable to prevent the use of this package scope except in the case of vetted and approved web components that the community would like to make available to everyone.

Scope would still be helpful to prevent naming conflicts with packages on NPM and/or Maven Central webjars.
Maybe we could add a new scope `@uportal-private` to prevent any conflicts and/or accidental publishes?
Thoughts? :thought_balloon:

Reply:
I like the idea of adding an additional scope as a safety measure. @ChristianMurphy Thanks for answering my questions. I appreciate it.

Reply:
Let's include the scope for now, and update as needed later.

Reply:
@uportal or @uportal-private?

Reply:
`@uportal` for now, we need to verify that `@uportal-private` is available.